### PR TITLE
feat: add complete example of infrastructure with aws

### PR DESCRIPTION
**Pull request by Henrrypg**
### Add Terraform Configuration for Basic AWS EKS Cluster

This PR introduces a set of Terraform files that provide a basic example or starting point for deploying a Kubernetes cluster on AWS using Elastic Kubernetes Service (EKS). The configuration includes:

- **VPC setup**: A Virtual Private Cloud (VPC) with customizable CIDR, public and private subnets, and NAT Gateway configurations.
- **EKS cluster**: A basic EKS cluster with customizable Kubernetes version, instance types, and auto-scaling options.
- **Node groups**: Configurable EC2 node groups with support for multiple instance types, disk size, and capacity (on-demand or spot).
- **Security and Networking**: SSH access configuration and security group setup for the EKS cluster and node groups.
- **Cluster Autoscaler support**: Optional enablement of cluster autoscaler.

### Key Files and Features:
- **`eks.tf`**: Core infrastructure setup for the EKS cluster.
- **`vpc.tf`**: Core infrastructure setup for the VPC.
- **`variables.tf`**: Parameters that allow customization of the cluster, such as instance types, cluster size, and networking options.
- **`outputs.tf`**: Exposes key outputs like cluster name and endpoint.
- **`provider.tf`**: Configures the AWS provider and region.

### How to use:
1. Customize the `variables.tf` or use a `values.auto.tfvars.json` file to adjust parameters like AWS region, instance types, and node group settings.
2. Run `terraform apply` to deploy the infrastructure.
> NOTE:  Follow the readme to further information.

This setup serves as a base or example for future enhancements and can be expanded with additional features or modules as needed.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Henrrypg!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.


#### :radio_button: Update the status of your PR

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.




#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/openedx-k8s-harmony-maintainers`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

**Comment by Henrrypg** [comment]
Hi, @MoisesGSalas @jfavellar90 

Can you look at this?

File: infra-examples/aws/eks/eks.tf
Comment (jfavellar90) [inline infra-examples/aws/eks/eks.tf:146]: Could we use a newer version of the [module](https://github.com/terraform-aws-modules/terraform-aws-iam)? This constraint only allows installing newer patches, keeping the same minor

File: infra-examples/aws/eks/eks.tf
Comment (jfavellar90) [inline infra-examples/aws/eks/eks.tf:163]: Could we use a newer version of the [module](https://github.com/terraform-aws-modules/terraform-aws-iam)? This constraint only allows installing newer patches, keeping the same minor

File: infra-examples/aws/eks/versions_override.tf
Comment (jfavellar90) [inline infra-examples/aws/eks/versions_override.tf:2]: Could we set the Terraform version to the latest minor available?

File: infra-examples/aws/eks/eks.tf
Comment (jfavellar90) [inline infra-examples/aws/eks/eks.tf:64]: I wonder if these rules are still required. I understand the SSH rule, but not sure about the other two. Could you please check what are the default rules defined in the module?

Comment (jfavellar90) [reply infra-examples/aws/eks/eks.tf:64]: @Henrrypg Yes, proceed with the deletion

**Comment by gabor-boros** [comment]
@bradenmacdonald Absolutely!

**Review by gabor-boros — APPROVED** [review]
This review is a bit delayed, sorry about that. I was thinking about this for a while now.

Overall, I think this PR is good as is -- after bumping the modules and providers to their latest version if applicable --, so I'm approving the changes here.

Besides that, this PR and the conversations we had in the working group during the meetings made me think if the file/directory structure is pointing towards the direction we want, i.e., having modules that can be used by community members in their Terraform scripts.

I would not like to block this PR from being merged on something that changed in the meantime, however, could you please open a GitHub issue as a "follow-up" to refactor the changes introduced here? I'll come up/propose a structure that in **my opinion** could be good, though that will be still up to the working group if we agree on that or not.

**Pull request merged**
